### DISCUSSION TITLE
How do I print data frames in for loop without using print function?

### DISCUSSION THREAD

**Discussion by barisguven** [quarto-dev/quarto-cli#6033] [Q&A] [answered]
### Description

I would like to print data frames generated in for loop as shown below. But Quarto does not print them neither when code chunk is run individually nor the qmd file is knitted to a html file. My objective is to get the data frames printed nicely as in kable style so that I can avoid using print function, which does not generate equally nice outcome.

````qmd
---
title: "testing data frame printing in for loop"
format: 
    html:
        df-print: kable
---

```{r}
columns <- c("mpg",  "cyl",  "disp", "hp")
mtcars[1:5, columns]
for (column in columns){
 mtcars[1:5, column]
}
```

**Comment by mcanouil** [comment, marked as answer]
You have to use (in your case) kable + print.
If you don't print the content of the loop, there is no way you get a display out of it (this is not limited to Quarto).

See:
- https://mickael.canouil.fr/posts/2023-03-05-quarto-auto-table-crossref/
- https://examples.quarto.pub/create-tabsets-panel-from-r-code/#resources

**Reply by cderv** [reply]
> You have to use (in your case) kable + print.

@barisguven More exactly prefered way would be to use child document or `knit_child()` so that  **knitr** does the correct printing of table for your in a knitting context. The two examples linked have such examples of usage. 

please try this approach which works better to output in a knitr chunk. Thanks !

**Reply by barisguven** [reply]
Thank you. @mcanouil's suggestion definitely works to get the multiple data frames or tables kable printed. It seems to me that the downside of using `knit_child()` is that you cannot view the data frames inside the Quarto document *as you do your analysis.* Perhaps you can still use for loop to generate a list of data frames and view or kable print them separately.

**Reply by cderv** [reply]
> It seems to me that the downside of using knit_child() is that you cannot view the data frames inside the Quarto document as you do your analysis.

Yes this exact. You won't be able to view it interactively. 

> Perhaps you can still use for loop to generate a list of data frames and but view or kable print them separately.

yes this is possible IMO. Same with plots.
